### ISSUE TITLE
OVR Platform does not have player information

### ISSUE THREAD

**Issue by yvasquezmonoku** [oculus-samples/Unreal-SharedSpaces#22]
I used to use "OculusPlatform" before and with "GetMyUserInfo " I was able to get the player's profile picture, but I upgraded to "OVR Platform" and I can't get the player's picture anymore as "GetMyUserInfo" doesn't exist anymore.
Is there any way to do it? or is this only exclusive to the previous "OculusPlatform" ?

**Comment by els-oculus** [comment]
Yes, in the newer OVRPlatform plugin, you have a number of functions used to retrieve the information about the current user.  The biggest difference between the newer OVRPlatform plugin and the previous iteration is that it is codegened to provide access to most of the OVR Platform C API calls.   By doing so, it is also much closer to that API.

The one you want is **User_Get()** in OVRPlatformRequest.h.    It is a full server request that retrieves what you need, assuming that you have allowed your app access to that information via Data Use Checkup.

    /**
     * Retrieve the user with the given ID. This might fail if the ID is invalid
     * or the user is blocked.
     * 
     * NOTE: Users will have a unique ID per application.
     * @param UserID - User ID retrieved with this application.
     */
    UFUNCTION(BlueprintCallable, meta = (Latent, LatentInfo = "LatentInfo", WorldContext = "WorldContextObject", ExpandEnumAsExecs = "OutExecs"), Category = "OvrPlatform|User")
    static void User_Get(
        UObject* WorldContextObject,
        EOvrRequestOutputPins& OutExecs,
        FLatentActionInfo LatentInfo,
        FOvrId UserID,
        FOvrUser& User,
        FString& ErrorMsg);


To use this function, you need the ID of the logged in user.  In OVRPlatformFunctions.h, you have **Platform_GetLoggedInUserID()** to just get the logged in user ID.

    // Returns the id of the currently logged in user, or a 0 id if there is none.
    UFUNCTION(BlueprintCallable, Category = "OvrPlatform|Platform")
    static FOvrId Platform_GetLoggedInUserID();


Now you have to be careful about another function found in OVRPlatformRequest.h: **User_GetLoggedInUser()**.   That function is OFFLINE, which means that it returns a subset of informations about the logged in user without placing a server call to retrieve all the data.

    /**
     * Retrieve the currently signed in user.  This call is available offline.
     * 
     * 
     * NOTE: This will not return the user's presence as it should always be 'online'
     * in your application.
     * 
     * 
     * NOTE: Users will have a unique ID per application.
     */
    UFUNCTION(BlueprintCallable, meta = (Latent, LatentInfo = "LatentInfo", WorldContext = "WorldContextObject", ExpandEnumAsExecs = "OutExecs"), Category = "OvrPlatform|User")
    static void User_GetLoggedInUser(
        UObject* WorldContextObject,
        EOvrRequestOutputPins& OutExecs,
        FLatentActionInfo LatentInfo,
        FOvrUser& User,
        FString& ErrorMsg);


The current implementation of the "Query Me" function in SharedSpaces makes use of those functions:

![query_logged_in_user](https://user-images.githubusercontent.com/61444568/174912382-037b215c-e96a-43d7-8290-9f6be67ade20.png)

**Comment by yvasquezmonoku** [comment]
I am trying to capture the name and url of the user's profile, but it does not return any data, using "PlatformGetLoggedinUser" it returns the user id but when using the User Get it does not return anything. I'm doing this from BP

**Comment by els-oculus** [comment]
A common issue would be that you did not authorize your app to read some data.  Check Data Use Checkup for your app id.
![duc](https://user-images.githubusercontent.com/61444568/176745750-0c9c1211-8175-4021-a454-d4cfbdedfd0b.png)

**Comment by yvasquezmonoku** [comment]
I got it to work, it seems that it should have a short delay. Thanks for the help

**Comment by els-oculus** [comment]
Oh, please be aware how those exit pins work.
- The top one (unnamed) is always executed without wait.   Your result won't be available yet.
- The "Success" or "Failure" pins will be exited when the result comes back.  Only when you exit through "Success" will you have the answer you want.
-

**Comment by yvasquezmonoku** [comment]
Thank you very much for responding, in my case it never came out in success or in fail. but by putting the delay before I called the function, it started working. I guess I should wait for the info to load.
